app/service/data_svc.py

```diff
@@ -142,8 +161,10 @@ def __init__(self):
         return phase_dict
 
     async def _insert_pack_phases(self, pack, phases, current_phase, adversary):
-        phases_new = await self._add_adversary_packs(pack)
-        if phases_new:
+        adv_pack = await self._grab_adversary(id=pack)
+        if adv_pack:
```

Comment:
doubling up on checking adv_pack is a code smell. you can adjust to make this cleaner.

Reply:
yea, I can do that, can probably combine into one conditional since arguments are executed in order if they evaluate true.